apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -450,6 +452,72 @@ type ApplyStrategy struct {
 	ServerSideApplyConfig *ServerSideApplyConfig `json:"serverSideApplyConfig,omitempty"`
 }
 
+// ConflictResolutionConfig describes the action used to resolve conflicts if the resource
+// to be placed already exists in the target cluster.
+type ConflictResolutionConfig struct {
+	// ActionType describes the type of the action used to resolve the conflict between the resource to be placed
+	// and the existing resource in the target cluster if they exist.
+	// +kubebuilder:default=AlwaysApply
+	// +kubebuilder:validation:Enum=AlwaysApply;Report;ApplyIfNoDiff;Recreate
+	// +optional
+	ActionType ConflictResolutionActionType `json:"actionType,omitempty"`
+
+	// DiffCalculateStrategy describes how to calculate the diff between the resource on
+	// the hub cluster and the existing resources on the member cluster.
+	// This strategy is only meaningful if the action type is ReportIfDiff or ApplyIfNoDiff.
+	// +optional
+	DiffCalculateStrategy DiffCalculateStrategy `json:"diffCalculateStrategy,omitempty"`
+}
+
+// ConflictResolutionActionType describes the type of the action to resolve the conflict
+// if the resource to be placed already exists in the target cluster.
+// +enum
+type ConflictResolutionActionType string
+
+const (
```

Comment:
The combos are:

* ReportIfDiff + HubFieldsOnly: complain if the apply would cause changes in the object, ignore changes in fields that are not mentioned by hub cluster resource
* ReportIfDiff + AllFields: complain if any field becomes inconsistent with the hub cluster resource
* ApplyIfNoDiff + HubFieldsOnly: complain if the apply would cause changes in the object, ignore changes in fields that are not mentioned by hub cluster resource (**duplicate**?)
* ApplyIfNoDiff + AllFieldsOnly: complain if any field becomes inconsistent with the hub cluster resource (**duplicate**?)
* AlwaysApply: overwrite any change in the fields mentioned by hub cluster resource
* Recreate: re-create if there's any change in the resource (*does this require diff option considerations?*)

Reply:
echo Chen's comment, 

what's the diff between ReportIfDiff and ApplyIfNoDiff?

They're same, if there is difference, we'll report the diff and stop applying?

Reply:
good point, changed the "ReportIfDiff" to "Report"